subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -92,16 +98,34 @@ private SnapshotHierarchy updateRootNotifyingWatchers(SnapshotHierarchy currentR
     }
 
     @Override
-    public void afterBuildStarted(boolean watchingEnabled) {
+    public void afterBuildStarted(boolean watchingEnabled, BuildOperationRunner buildOperationRunner) {
         reasonForNotWatchingFiles = null;
         rootReference.update(currentRoot -> {
             if (watchingEnabled) {
-                SnapshotHierarchy newRoot = handleWatcherRegistryEvents(currentRoot, "since last build");
-                if (watchRegistry == null) {
-                    newRoot = startWatching(newRoot);
-                }
-                printStatistics(newRoot, "retained", "since last build");
-                return newRoot;
+                return buildOperationRunner.call(new CallableBuildOperation<SnapshotHierarchy>() {
+                    @Override
+                    public SnapshotHierarchy call(BuildOperationContext context) {
+                        SnapshotHierarchy newRoot = currentRoot;
+                        boolean startedWatching = false;
+                        FileWatcherRegistry.FileWatchingStatistics statistics = watchRegistry == null
+                            ? null
+                            : watchRegistry.getAndResetStatistics();
+
+                        newRoot = handleWatcherRegistryEvents(newRoot, "since last build", statistics);
+                        if (watchRegistry == null) {
+                            startedWatching = true;
+                            context.setStatus("Starting file system watching");
+                            newRoot = startWatching(newRoot);
+                        }
+                        context.setResult(new BuildStartedWithFileSystemWatchingEnabled(startedWatching, statistics, newRoot));
```

Comment:
Don't we want to capture the whole of `afterBuildStarted()`? IIRC, we said we want to capture whether FSW was enabled in the result object.

Reply:
Capturing the whole of `afterBuildStarted` now.